cfgov/v1/tests/models/test_related_posts.py

```diff
@@ -0,0 +1,212 @@
+        helpers.save_new_page(self.archive_events_parent)
+
+        # set up children of the parent pages and give them some tags
+        # and some categories
+
+        self.blog_child1 = AbstractFilterPage(title='blog child 1', date_published=dt.date(2016, 9, 1))
+        self.blog_child1.tags.add('tag 1')
+        self.blog_child1.categories.add(CFGOVPageCategory(name='info-for-consumers'))
+
+        self.blog_child2 = AbstractFilterPage(title='blog child 2', date_published=dt.date(2016, 9, 2))
+        self.blog_child2.tags.add('tag 2')
+        self.blog_child2.categories.add(CFGOVPageCategory(name='policy_compliance'))
+
+        self.newsroom_child1 = AbstractFilterPage(title='newsroom child 1', date_published=dt.date(2016, 9, 2))
+        self.newsroom_child1.tags.add('tag 1')
+        self.newsroom_child1.tags.add('tag 2')
+        self.newsroom_child1.categories.add(CFGOVPageCategory(name='op-ed'))
+
+        self.newsroom_child2 = AbstractFilterPage(title='newsroom child 2', date_published=dt.date(2016, 9, 3))
+        self.newsroom_child2.tags.add('tag 2')
+        self.newsroom_child2.categories.add(CFGOVPageCategory(name='some-other-category'))
+
+        self.events_child1 = AbstractFilterPage(title='events child 1', date_published=dt.date(2016, 9, 7))
+        self.events_child1.tags.add('tag 1')
+
+        self.events_child2 = AbstractFilterPage(title='events child 2', date_published=dt.date(2016, 9, 5))
+        self.events_child2.tags.add('tag 2')
+
+        helpers.save_new_page(self.blog_child1, self.blog_parent)
+        helpers.save_new_page(self.blog_child2, self.blog_parent)
+        helpers.save_new_page(self.newsroom_child1, self.newsroom_parent)
+        helpers.save_new_page(self.newsroom_child2, self.newsroom_parent)
+        helpers.save_new_page(self.events_child1, self.events_parent)
+
+        # mock a stream block that dictates how to retrieve the related posts
+        # note that because of the way that the related_posts_category_lookup function
+        # works i.e. by consulting a hard-coded object, the specific_categories
+        # slot of the dict has to be something that it can actually find.
+
+        self.block = mock.MagicMock()
+        self.block.block_type = 'related_posts'
+        self.block.value = dict({
+            'limit': 3,
+            'show_heading': True,
+            'header_title': 'Further reading',
+            'relate_posts': False,
+            'relate_newsroom': False,
+            'relate_events': False,
+            'specific_categories': []
+        })
+
+    def tearDown(self):
+
+        # you don't need to delete the children independently,
+        # you can just delete the parent and it cascades
+
+        self.blog_parent.delete()
+        self.newsroom_parent.delete()
+        self.events_parent.delete()
+        self.archive_events_parent.delete()
+
+        self.page_with_authors.delete()
+
+    def test_related_posts_blog(self):
```

Comment:
Would it be possible to make this test name more descriptive, or describe it in a docstring? ditto for the rest of the tests. There are multiple assertions here and I'm not sure which each one is testing. e.g. what is `self.assertEqual(len(related_posts['Blog']), 2)` testing?  why do we expect it to be 2?